### DISCUSSION TITLE
2.0 Migration Guide?

### DISCUSSION THREAD

**Discussion by IsaacMarovitz** [Alexandre-Fernandez/astro-i18n#65] [General]
I have a site that is localised with `astro-i18n` 1.8.1. How do I go about upgrading this site to 2.x? Is there a guide I can follow?

My site's full source code is available here: https://github.com/Whisky-App/WhiskySite

**Comment by Alexandre-Fernandez** [comment]
Hi, there's no guide however you do not have a lot of changes to do. Mostly some renamings and moving your translations into the common group (if you do not want to use the new scoping features). 
To keep the library server-side only, modify the new `run` option to `"server"`. Other than that you follow the new README and check the test repo https://github.com/Alexandre-Fernandez/astro-i18n/tree/main/apps/astro for reference.

**Reply by IsaacMarovitz** [reply]
I've got a couple of questions:
1. Is it still possible to have languages separated into individual files like my current configuration? My website is localised through Crowdin, which relies on them being in this format.
2. When trying to run `astro dev` or `astro build` I run into the following error:
```

SyntaxError: Cannot use 'import.meta' outside a module
    at internalCompileFunction (node:internal/vm:77:18)
    at wrapSafe (node:internal/modules/cjs/loader:1288:20)
    at Module._compile (node:internal/modules/cjs/loader:1340:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at cjsLoader (node:internal/modules/esm/translators:345:17)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:294:7)
    at ModuleJob.run (node:internal/modules/esm/module_job:218:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:329:24)
```

**Reply by Alexandre-Fernandez** [reply]
> Is it still possible to have languages separated into individual files like my current configuration? My website is localised through Crowdin, which relies on them being in this format.

Yes you can have them in that format, see https://github.com/Alexandre-Fernandez/astro-i18n#get-started.

> When trying to run astro dev or astro build I run into the following error:

Where can I test that, please open an issue with a reproduction repo so I can check if it comes from `astro-i18n` and fix it.

**Reply by Nkay** [reply]
@Alexandre-Fernandez thanks a lot for creating the v2!
Having types will make this far better :)

I've got the same problem; here is more information:
"(node:23024) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
node_modules\astro-i18n\dist\src\index.js:12


SyntaxError: Cannot use 'import.meta' outside a module
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1178:20)
    at Module._compile (node:internal/modules/cjs/loader:1220:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
"
Node v18.16.1


I tried to reproduce it in an empty project to no avail, it works in a new project.
I created a public repo with a minimal configuration that has this problem now. I will create an issue. Thanks!

**Reply by Alexandre-Fernandez** [reply]
did you fix the problem ?

**Reply by Nkay** [reply]
Hi @Alexandre-Fernandez, thanks for asking. It fixed that problem, and it created a new one ;)
I'm just in the finalization of a new minimal reproduction repo and will create an Issue. Thanks for the support!

**Reply by Nkay** [reply]
Additionally, for anyone who could find it useful:

 - removed the astro-i18n package and readded it according to the install so I get the new default config, which I then adapted
 - I had to remove the middleware from astro.config.mjs
 - Used the new HrefLang - Tool from the Readme instead of the previously bundled HrefLang-Component
 - t() - Calls with a nonexisting key return the key instead of an empty string now
 - astroI18n.langCode does not exist anymore - astroI18n.locale is the new one
 - changed the location of the translation files - in my case from /src/i18n/de/translation.json to /src/i18n/common/de.json
 - Removed all calls to astroI18n.init(Astro);

Guess that was about it, just that currently the translations out of .ts files do not work for me - that Issue #68: https://github.com/Alexandre-Fernandez/astro-i18n/issues/68
